sound/soc/sof/intel/hda-ipc.c

```diff
@@ -234,19 +234,21 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 {
```

Comment:
please fix your commit message. You *cannot* use a 4-line sentence. Break it up in smaller sentences with simple subject-verb-complement structure.

ASoC: SOF: Intel: hda-ipc: save ipc irq_status only for IPC IRQ

We are using a shared IRQ for IPC and stream IOC interrupts. We need to check the interrupt source before storing the status...

The rest makes no sense...
 otherwise, the previously stored one may be
modified(to 0 usually) when a consequent IOC interrupt arrives.
???

This rather looks like the variable used in the IRQ thread could be modified by the first part of the interrupt? Your explanations are really confusing.

From now on, please assume you have to push the patches directly to alsa-devel. Would you seriously send something for review with that level of ambiguity in the wording?

Reply:
> please fix your commit message. You _cannot_ use a 4-line sentence. Break it up in smaller sentences with simple subject-verb-complement structure.
> 
> ASoC: SOF: Intel: hda-ipc: save ipc irq_status only for IPC IRQ
> 
> We are using a shared IRQ for IPC and stream IOC interrupts. We need to check the interrupt source before storing the status...
> 
> The rest makes no sense...
> otherwise, the previously stored one may be
> modified(to 0 usually) when a consequent IOC interrupt arrives.
> ???
> 
> This rather looks like the variable used in the IRQ thread could be modified by the first part of the interrupt? Your explanations are really confusing.

Thanks for @xiulipan's explaining, yes, that's the case.
 
Sorry for making confusing explanations, that's not my intention, @plbossart I think you have understood how the issue happens and why we need this fix here, right? If so, let me focus on refining commit messages though it is always a challenge to me. 

> 
> From now on, please assume you have to push the patches directly to alsa-devel. Would you seriously send something for review with that level of ambiguity in the wording?